src/huggingface_hub/serialization/_torch.py

```diff
@@ -349,6 +354,336 @@ def split_torch_state_dict_into_shards(
     )
 
 
+# LOADING
+
+
+def load_torch_model(
+    model: "torch.nn.Module",
+    checkpoint_path: Union[str, os.PathLike],
```

Comment:
(unrelated to this PR) I used to put `Union[str, Path]` in a lot of places in `huggingface_hub` but agree `Union[str, os.PathLike]` is more appropriate. Would be cool to harmonize this at some point (not urgent)